### ISSUE TITLE
Focal loss calculation

### ISSUE THREAD

**Issue by james-nvidia** [NVIDIA/retinanet-examples#182]
Hey,
I'd like to know wether the focal loss is calculated on the test dataset or the validation one ? 

I also would like to modify this focal loss to tune it for my case. I want to add a term taking into account the precision of the center of my predicted bounding boxes. In what file should I add this term ? 

Thank you

_Originally posted by @VincentDuf in https://github.com/NVIDIA/retinanet-examples/issues/158#issuecomment-617230204_

**Comment by james-nvidia** [comment]
The focal loss is calculated in `loss.py`
https://github.com/NVIDIA/retinanet-examples/blob/master/retinanet/loss.py

The loss is calculated on the training dataset, as specified by `--images` and `--annotations`

**Comment by VincentDuf** [comment]
Ok thank you.
Is there a way to add a "focal loss" on the validation dataset ? 

Concerning the loss if I want to add an extra term, should I insert it in the line 18 when the forward function return the result ? 

Thanks
